File: ruby/lib/coding_interview/domain/app_constants.rb
require "bigdecimal"
require_relative "stock"
require_relative "stock_symbol"

module CodingInterview
  module Domain
    module AppConstants
      CASH_RATE = BigDecimal("0.05")
      MIN_OPERATION_AMOUNT = BigDecimal("10000")

      SUPPORTED_SYMBOLS = [StockSymbol::Toyopa, StockSymbol::Somy].freeze

      INITIAL_PRICES = {
        StockSymbol::Toyopa => BigDecimal("4.2135"),
        StockSymbol::Somy => BigDecimal("1.2345")
      }.freeze

      INITIAL_PORTFOLIO = Portfolio.new([
        PortfolioItem.new(StockSymbol::Toyopa, BigDecimal("0.40")),
        PortfolioItem.new(StockSymbol::Somy, BigDecimal("0.60"))
      ])
    end
  end
end

File: java17/src/main/java/folio/codinginterview/domain/AppConstants.java
package folio.codinginterview.domain;

import java.math.BigDecimal;
import java.util.List;

public final class AppConstants {
    private AppConstants() {}

    public static final BigDecimal CASH_RATE = new BigDecimal("0.05");

    public static final BigDecimal MIN_OPERATION_AMOUNT = new BigDecimal(10000);

    public static final List<StockSymbol> SUPPORTED_SYMBOLS = List.of(StockSymbol.Toyopa, StockSymbol.Somy);

    public static final Portfolio INITIAL_PORTFOLIO = new Portfolio(List.of(
            new PortfolioItem(StockSymbol.Toyopa, new BigDecimal("0.40")),
            new PortfolioItem(StockSymbol.Somy, new BigDecimal("0.60"))
    ));
}
